workflows/funcscan.nf

```diff
@@ -64,98 +54,106 @@ workflow FUNCSCAN {
+                if (fasta != [] && fasta.isEmpty()) {
+                    log.warn("[nf-core/funcscan] Sample ${meta.id} does not have contigs longer than ${params.bgc_mincontiglength} bp. Will not be screened for BGCs.")
+                }
+                !fasta.isEmpty()
             }
-        ch_versions = ch_versions.mix( SEQKIT_SEQ.out.versions )
-    } else {
-        ch_input_for_annotation = ch_intermediate_input.fastas.map { meta, fasta, protein, gbk ->  [ meta, fasta ] }
+        ch_versions = ch_versions.mix(SEQKIT_SEQ_LENGTH.out.versions)
+    }
+    else {
+        ch_input_for_annotation = ch_intermediate_input.fastas.map { meta, fasta, protein, gbk -> [meta, fasta] }
     }
 
     /*
         ANNOTATION
     */
 
     // Some tools require annotated FASTAs
-    if ( ( params.run_arg_screening && !params.arg_skip_deeparg ) || ( params.run_amp_screening ) || ( params.run_bgc_screening ) ) {
-        ANNOTATION( ch_input_for_annotation )
-        ch_versions = ch_versions.mix( ANNOTATION.out.versions )
+    if ((params.run_arg_screening && !params.arg_skip_deeparg) || params.run_amp_screening || params.run_bgc_screening) {
+        ANNOTATION(ch_input_for_annotation)
+        ch_versions = ch_versions.mix(ANNOTATION.out.versions)
 
         ch_new_annotation = ch_input_for_annotation
-                                .join( ANNOTATION.out.faa )
-                                .join( ANNOTATION.out.gbk )
-
-    } else {
+            .join(ANNOTATION.out.faa)
+            .join(ANNOTATION.out.gbk)
+    }
+    else {
         ch_new_annotation = ch_intermediate_input.fastas
     }
 
     // Mix back the preannotated samples with the newly annotated ones
     ch_prepped_input = ch_new_annotation
-                        .filter { meta, fasta, faa, gbk -> meta.category != 'long' }
-                        .mix( ch_intermediate_input.preannotated )
-                        .multiMap {
-                            meta, fasta, faa, gbk ->
-                                fastas: [meta, fasta]
-                                faas: [meta, faa]
-                                gbks: [meta, gbk]
-                        }
-
-    if ( params.run_bgc_screening ){
-
-        ch_prepped_input_long =  ch_new_annotation
-                                    .filter { meta, fasta, faa, gbk -> meta.category == 'long'}
-                                    .mix( ch_intermediate_input.preannotated )
-                                    .multiMap {
-                                        meta, fasta, faa, gbk ->
-                                            fastas: [meta, fasta]
-                                            faas: [meta, faa]
-                                            gbks: [meta, gbk]
-                                    }
+        .filter { meta, fasta, faa, gbk -> meta.category != 'long' }
+        .mix(ch_intermediate_input.preannotated)
+        .multiMap { meta, fasta, faa, gbk ->
+            fastas: [meta, fasta]
```

Comment:
```suggestion
        .multiMap { meta, fna, faa, gbk ->
            fastas: [meta, fna]
```
This is the contigs I assume. In that case, this suggestion would be in line at least with Prokka's file naming.

Reply:
Yes, throughout the pipeline we use `fasta` for contigs and `fna` for annotated files (in line with `faa`, `gbk`). It's just an unwritten law. In the above case, `fasta` contains the contigs from the samplesheet, so I would leave this as is.

Reply:
Wait, no it does not! Contains the annotated files (ending in `fna`), you're right! Thanks for spotting

Reply:
Sorry for confusion, I checked again with a fresh brain: No, we leave as is because "fasta" is the non-annotated contigs. The downstream tools use only "gbk" and "faa" files from the annotation step, and no "fna" files.